api/provider/aws/environment_update.go

```diff
@@ -18,24 +19,23 @@ import (
 func (e *EnvironmentProvider) Update(environmentID string, req models.UpdateEnvironmentRequest) error {
 	fqEnvironmentID := addLayer0Prefix(e.Config.Instance(), environmentID)
 
-	if req.MinScale != nil || req.MaxScale != nil {
-		autoScalingGroupName := fqEnvironmentID
-		asg, err := e.readASG(autoScalingGroupName)
+	if req.Scale != nil {
+		tags, err := e.TagStore.SelectByTypeAndID("environment", environmentID)
 		if err != nil {
 			return err
 		}
 
-		minSize := aws.Int64Value(asg.MinSize)
-		if req.MinScale != nil {
-			minSize = int64(*req.MinScale)
+		environmentType := ""
+		if tag, ok := tags.WithKey("type").First(); ok {
+			environmentType = tag.Value
 		}
 
-		maxSize := aws.Int64Value(asg.MaxSize)
-		if req.MaxScale != nil {
-			maxSize = int64(*req.MaxScale)
+		if environmentType != models.EnvironmentTypeStatic {
+			return fmt.Errorf("updating dynamic environment's scale isn't supported")
```

Comment:
I'm pretty sure there is an `InvalidRequest` error code that we could use for this, e.g. `return errors.New(errors.InvalidRequest, "updating...")`. That'll be helpful since that will let the `controller` pkg know to return a 400 status code rather than a 500.